dataland-internal-storage/src/test/kotlin/org/dataland/datalandinternalstorage/services/DatabaseDataStoreTest.kt

```diff
@@ -0,0 +1,61 @@
+package org.dataland.datalandinternalstorage.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import jakarta.transaction.Transactional
+import org.dataland.datalandbackend.openApiClient.api.NonPersistedDataControllerApi
+import org.dataland.datalandinternalstorage.DatalandInternalStorage
+import org.dataland.datalandinternalstorage.entities.DataItem
+import org.dataland.datalandinternalstorage.repositories.DataItemRepository
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.springframework.amqp.AmqpException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.jdbc.EmbeddedDatabaseConnection
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.context.SpringBootTest
+import java.lang.IllegalArgumentException
+
+@SpringBootTest(classes = [DatalandInternalStorage::class])
+@AutoConfigureTestDatabase(connection = EmbeddedDatabaseConnection.H2)
+@Transactional
+class DatabaseDataStoreTest(
+    @Autowired val objectMapper: ObjectMapper,
+) {
+    val mockDataItemRepository: DataItemRepository = mock(DataItemRepository::class.java)
+    val mockCloudEventMessageHandler: CloudEventMessageHandler = mock(CloudEventMessageHandler::class.java)
+    val mockNonPersistedDataClient: NonPersistedDataControllerApi = mock(NonPersistedDataControllerApi::class.java)
+    val databaseDataStore = DatabaseDataStore(
+        mockDataItemRepository, mockCloudEventMessageHandler,
+        mockNonPersistedDataClient, objectMapper,
+    )
+    val dataId = "TestDataId"
+    val data = "TestDataForTestDataId"
+    val correlationId = "TestCorrelationId"
+
+    @Test
+    fun `check that an exception is thrown during insertion when the input is invalid`() {
+        `when`(mockDataItemRepository.save(DataItem(dataId, objectMapper.writeValueAsString(data)))).thenThrow(
+            IllegalArgumentException::class.java,
+        )
+        assertThrows<IllegalArgumentException> {
+            databaseDataStore.insertDataAndSendMessage(dataId, data, correlationId)
+        }
+    }
+
+    @Test
+    fun `check that an exception is thrown when sending a success notification to message queue fails`() {
+        `when`(
+            mockCloudEventMessageHandler.buildCEMessageAndSendToQueue(
+                dataId, "Data successfully stored", correlationId, "stored_queue",
```

Comment:
The reasoning in https://github.com/d-fine/Dataland/pull/180/files#r1114799640 applies here as well.

Reply:
This one was also removed because exception thrown immediately goes out. Basically this test checked if buildCEMessageAndSendToQueue was called

Reply:
ok